packages/cli/test/integration/publicApi/credentials.test.ts

```diff
@@ -0,0 +1,313 @@
+	let ownerShell = await testDb.createUserShell(globalOwnerRole);
+	ownerShell = await testDb.addApiKey(ownerShell);
+
+	const authOwnerAgent = utils.createAgent(app, {
+		apiPath: 'public',
+		version: 1,
+		auth: true,
+		user: ownerShell,
+	});
+
+	await Promise.all(
+		INVALID_PAYLOADS.map(async (invalidPayload) => {
+			const response = await authOwnerAgent.post('/credentials').send(invalidPayload);
+			expect(response.statusCode === 400 || response.statusCode === 415).toBe(true);
+		}),
+	);
+});
+
+test('POST /credentials should fail with missing encryption key', async () => {
+	const mock = jest.spyOn(UserSettings, 'getEncryptionKey');
+	mock.mockRejectedValue(new Error(RESPONSE_ERROR_MESSAGES.NO_ENCRYPTION_KEY));
+
+	let ownerShell = await testDb.createUserShell(globalOwnerRole);
+	ownerShell = await testDb.addApiKey(ownerShell);
+
+	const authOwnerAgent = utils.createAgent(app, {
+		apiPath: 'public',
+		version: 1,
+		auth: true,
+		user: ownerShell,
+	});
+
+	const response = await authOwnerAgent.post('/credentials').send(credentialPayload());
+
+	expect(response.statusCode).toBe(500);
+
+	mock.mockRestore();
+});
+
+test('POST /credentials should ignore ID in payload', async () => {
+	let ownerShell = await testDb.createUserShell(globalOwnerRole);
+	ownerShell = await testDb.addApiKey(ownerShell);
+
+	const authOwnerAgent = utils.createAgent(app, {
+		apiPath: 'public',
+		version: 1,
+		auth: true,
+		user: ownerShell,
+	});
+
+	const firstResponse = await authOwnerAgent
+		.post('/credentials')
+		.send({ id: '8', ...credentialPayload() });
+
+	expect(firstResponse.body.id).not.toBe('8');
+
+	const secondResponse = await authOwnerAgent
+		.post('/credentials')
+		.send({ id: 8, ...credentialPayload() });
+
+	expect(secondResponse.body.id).not.toBe(8);
+});
+
+test('DELETE /credentials/:id should delete owned cred for owner', async () => {
+	let ownerShell = await testDb.createUserShell(globalOwnerRole);
+	ownerShell = await testDb.addApiKey(ownerShell);
+
+	const authOwnerAgent = utils.createAgent(app, {
+		apiPath: 'public',
+		version: 1,
+		auth: true,
+		user: ownerShell,
+	});
+
+	const savedCredential = await saveCredential(credentialPayload(), { user: ownerShell });
+
+	const response = await authOwnerAgent.delete(`/credentials/${savedCredential.id}`);
+
+	expect(response.statusCode).toBe(200);
+
+	const { name, type } = response.body;
+
+	expect(name).toBe(savedCredential.name);
+	expect(type).toBe(savedCredential.type);
+
+	const deletedCredential = await Db.collections.Credentials!.findOne(savedCredential.id);
+
+	expect(deletedCredential).toBeUndefined(); // deleted
+
+	const deletedSharedCredential = await Db.collections.SharedCredentials!.findOne();
+
+	expect(deletedSharedCredential).toBeUndefined(); // deleted
+});
+
+test('DELETE /credentials/:id should delete non-owned cred for owner', async () => {
+	let ownerShell = await testDb.createUserShell(globalOwnerRole);
+	ownerShell = await testDb.addApiKey(ownerShell);
```

Comment:
Utils.createUser() should replace both lines. It creates a member and automatically assigns an API key

Reply:
disagree